lib/pbench/server/indexing_tarballs.py

```diff
@@ -425,13 +439,33 @@ def sigquit_handler(*args):
 
                         if sigquit_interrupt[0]:
                             break
+                        if sighup_interrupt[0]:
+                            status, new_tb = self.collect_tb()
+                            if status == 0 and new_tb:
```

Comment:
Here's an awkward question:  if `new_tb` evaluates to `False`, then we won't do the superset test and, more importantly, we won't update `tb_deque`.  The principal reason why `new_tb` would evaluate to `False` (accompanied by a "success" status) would be if someone went in and deleted all of the two-be-indexed datasets, presumably in an attempt to abort indexing.  So, what is going to happen next is that the tarball open (or something similar) will fail...which seems ugly.

This is a long-winded setup to ask, why are we testing `new_tb` at line 444?  Wouldn't it be better to just proceed based on the value of `status`?  (If `new_tb` turns out to be empty, won't proceeding make happen what we want to have happen?)

(Riya, I leave it up to you to decide how to address this -- you could fix it in this PR, do it in another, or just capture it as an issue.)

Reply:
> Here's an awkward question: if `new_tb` evaluates to `False`, then we won't do the superset test and, more importantly, we won't update `tb_deque`. The principal reason why `new_tb` would evaluate to `False` (accompanied by a "success" status) would be if someone went in and deleted all of the two-be-indexed datasets, presumably in an attempt to abort indexing. So, what is going to happen next is that the tarball open (or something similar) will fail...which seems ugly.

@webbnh The reason that new_tb would evaluate as false is when there aren't any tarballs that need indexing("TO-INDEX" folder is empty). Maybe l.128 through l.130 can clear something up.


>


>
> This is a long-winded setup to ask, why are we testing `new_tb` at line 444? Wouldn't it be better to just proceed based on the value of `status`? (If `new_tb` turns out to be empty, won't proceeding make happen what we want to have happen?)

we could have proceeded just with status value but if there would have been no tarballs then all the checkings and replacing inside condition will all just not make sense.
if 'new_tb' will turn out to be empty then 'tb_deque' will run as it should and will end as soon as 'tb_deque' is empty.

Am I missing something?

Reply:
If we're running this code, then `tb_deque` had at least one entry in it -- let's assume that it had several -- and something sent a `SIGHUP` which made us call `collect_tb()`.  Now, for the sake of argument, suppose that the TO-INDEX folder is now empty (somehow...), which means that `new_tb` will be empty, but, for the sake of argument, suppose that `tb_deque` is _not_ empty (because we've indexed only the first item of the several).

At this point, we have an interesting choice.  We _could_ update `tb_deque` to be empty, which I think makes good sense, because there is presumably nothing in the TO-INDEX folder, now.  This will (I think) allow indexing to conclude in an orderly fashion.  The alternative, as currently coded, is to ignore the update, do nothing to `tb_deque`, and, on the next iteration of the loop, discover that the next tarball is nowhere to be found.

> Am I missing something?

Sleep?  :-D

Reply:
(Or, am _I_ missing something??  :-) )

Reply:
It is quite an exceptional condition but there is no harm in making it work. As, either both(tb_deque and new_tb) should be empty or both should be containing the same elements.

> Sleep? :-D

haha

Reply:
but yeah there is one more thing it will also add the somehow deleted tarballs in the missing log after not passing the superset condition.

Reply:
fixed

Reply:
> it will also add the somehow deleted tarballs in the missing log after not passing the superset condition.

And that's exactly what I think we want:  the tarballs were there when we started, and they are not there now, and they didn't get indexed, and that's weird and worth logging.